servertest/servertest.go

```diff
@@ -0,0 +1,16 @@
+package servertest
+
+import (
+	"net"
+)
+
+// NewLocalListener returns a system-chosen port on the local loopback interface, for use in end-to-end HTTP tests
+func NewLocalListener() (net.Listener, error) {
```

Comment:
Why not include `NewLocalListener` inside the `server` package? Is this intended to be a utility that developers can use for their own testing purposes?

If so, adding some documentation to the toolkit README that describes the `server` and `servertest` packages would be helpful (once the PR is closer to getting merged).

Reply:
good callout on updating the README.  I copied what Go did with their [httptest package](https://golang.org/pkg/net/http/httptest/) to get this. It's a test helper thing, so it shouldn't live in the non-test `server` package.